crates/anvil/src/lib.rs

```diff
@@ -85,15 +89,15 @@ pub mod cmd;
 /// # use anvil::NodeConfig;
 /// # async fn spawn() {
 /// let config = NodeConfig::default();
-/// let (api, handle) = anvil::spawn(config).await;
+/// let (api, _engine_api, handle) =anvil::spawn(config).await;
 ///
 /// // use api
 ///
 /// // wait forever
 /// handle.await.unwrap();
 /// # }
 /// ```
-pub async fn spawn(mut config: NodeConfig) -> (EthApi, NodeHandle) {
+pub async fn spawn(mut config: NodeConfig) -> (EthApi, EngineApi, NodeHandle) {
```

Comment:
here we should keep the original spawn function, just so that we don't break too much

but we definitely want a new one that can also return the EngineApi.

In other words, since this function accepts the NodeConfig, which determines whether the EngineApi is returned, we want a function interface like this. so this function should be renamed, we keep `spawn` that then calls it (perhaps we call it `spawn_node` or `spawn_all`)

another solution would be to give the NodeHandle a clone of both, EthApi and Option<EngineApi> then we don't have the issue at all.